projects/super_res/model/autoreg_diffusion_mod.py

```diff
@@ -0,0 +1,1442 @@
+                        loss = self.model(lres, hres)
+                        loss = loss / self.gradient_accumulate_every
+                        total_loss += loss.item()
+
+                    self.accelerator.backward(loss)
+
+                accelerator.clip_grad_norm_(self.model.parameters(), 1.0)
+                pbar.set_description(f'loss: {total_loss:.4f}')
+
+                #self.writer.add_scalar("loss", total_loss, self.step)
+                accelerator.log({"loss": total_loss}, step = self.step)
+
+                accelerator.wait_for_everyone()
+
+                self.opt.step()
+                self.opt.zero_grad()
+
+                accelerator.wait_for_everyone()
+
+                self.step += 1
+                if accelerator.is_main_process:
+                    self.ema.to(device)
+                    self.ema.update()
+
+                    if self.step != 0 and self.step % self.save_and_sample_every == 0:
+                        self.ema.ema_model.eval()
+
+                        with torch.no_grad():
+                            
+                            for i, batch in enumerate(self.val_dl):
+                                
+                                lres = batch['LR'].to(device)
+                                hres = batch['HR'].to(device)
+                                
+                                if i >= self.val_num_of_batch:
+                                    break
+
+                                num_samples = 5
+                                num_videos_per_batch = 2
+                                num_frames = 5
+                                img_size = 384
+                                img_channels = 1
+
+                                truth = np.zeros((1, num_videos_per_batch, num_frames, img_channels, img_size, img_size), dtype = 'uint8')
+                                pred = np.zeros((num_samples, 1, num_videos_per_batch, num_frames, img_channels, img_size, img_size), dtype = 'uint8')
+                                truth[0,:,:,:,:,:] = (hres[:,2:,:,:,:].repeat(1,1,1,1,1).cpu().numpy()*255).astype(np.uint8)
+                                for k in range(num_samples):
+                                    videos, base, res, flows = self.ema.ema_model.sample(lres)
+                                    pred[k,0,:,:,:,:] = (videos.clamp(0.0, 1.0)[:,:,0:1,:,:].repeat(1,1,1,1,1).detach().cpu().numpy()*255).astype(np.uint8)
+                                
+                                crps_index = calculate_crps(truth, pred, num_samples, num_videos_per_batch, num_frames, img_channels, img_size)
+                                psnr_index = piq.psnr(hres[:,2:,0:1,:,:], videos.clamp(0.0, 1.0)[:,:,0:1,:,:], data_range=1., reduction='none')
+                                
+                                accelerator.log({"true_high": wandb.Video((hres[:,2:,0:1,:,:].repeat(1,1,3,1,1).cpu().numpy()*255).astype(np.uint8))}, step=self.step)
```

Comment:
Let's take this opportunity to change these names into the interpretable ones that you are using your wandb [workspace](https://wandb.ai/ai2cm/vsr-orig-autoreg-hres/runs/hpc6lshz?workspace=user-prakhars). Does this sound reasonable?

Reply:
We can easily edit them in WandB, the only reason I didn't change it here was because the code becomes unnecessarily wordy